trinity/main.py

```diff
@@ -220,6 +256,10 @@ def trinity_boot(args: Namespace,
         # simply uses 'kill' to send a signal to the main process, but also because they will
         # perform a non-gracefull shutdown if the process takes too long to terminate.
         logger.info('Keyboard Interrupt: Stopping')
+        plugin_manager.shutdown()
+        networking_endpoint.stop()
+        main_endpoint.stop()
+        event_bus.stop()
```

Comment:
Note to self: `event_bus.stop()` should probably just call `stop()` on every `endpoint`